providers/hns/hns_roce_u_verbs.c

```diff
@@ -272,25 +293,42 @@ static int hns_roce_alloc_cq_buf(struct hns_roce_cq *cq)
 }
 
 static int exec_cq_create_cmd(struct ibv_context *context,
-			      struct hns_roce_cq *cq, int cqe,
-			      struct ibv_comp_channel *channel, int comp_vector)
+			      struct hns_roce_cq *cq,
+			      struct ibv_cq_init_attr_ex *attr)
 {
+	struct hns_roce_create_cq_ex_resp resp_ex = {};
 	struct hns_roce_ib_create_cq_resp *resp_drv;
+	struct hns_roce_create_cq_ex cmd_ex = {};
 	struct hns_roce_create_cq_resp resp = {};
 	struct hns_roce_ib_create_cq *cmd_drv;
 	struct hns_roce_create_cq cmd = {};
+	bool use_ex;
 	int ret;
 
-	cmd_drv = &cmd.drv_payload;
-	resp_drv = &resp.drv_payload;
+	use_ex = attr->flags & CREATE_CQ_EX_FLAGS;
+
+	if (use_ex) {
+		cmd_drv = &cmd_ex.drv_payload;
+		resp_drv = &resp_ex.drv_payload;
+	} else {
+		cmd_drv = &cmd.drv_payload;
+		resp_drv = &resp.drv_payload;
+	}
 
 	cmd_drv->buf_addr = (uintptr_t)cq->buf.buf;
 	cmd_drv->db_addr = (uintptr_t)cq->db;
 	cmd_drv->cqe_size = (uintptr_t)cq->cqe_size;
 
-	ret = ibv_cmd_create_cq(context, cqe, channel, comp_vector,
-				&cq->ibv_cq, &cmd.ibv_cmd, sizeof(cmd),
-				&resp.ibv_resp, sizeof(resp));
+	if (use_ex)
+		ret = ibv_cmd_create_cq_ex(context, attr, &cq->verbs_cq,
+					   &cmd_ex.ibv_cmd, sizeof(cmd_ex),
+					   &resp_ex.ibv_resp, sizeof(resp_ex));
+	else
+		ret = ibv_cmd_create_cq(context, attr->cqe, attr->channel,
```

Comment:
As we just discussed on mlx5, simply always call ibv_cmd_create_cq_ex, it now internally does this logic.